sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -407,10 +429,7 @@ object ResolveDefaultColumns extends QueryErrorsBase
       val defaultValue: Option[String] = field.getExistenceDefaultValue()
       defaultValue.map { text: String =>
         val expr = try {
-          val expr = analyze(field, "", EXISTS_DEFAULT_COLUMN_METADATA_KEY)
-          expr match {
-            case _: ExprLiteral | _: Cast => expr
-          }
+          analyzeExistingDefault(field, text)
```

Comment:
unit testing idea: we could try to stub out the catalog manager with something that always throws errors, if we wanted to be extra careful. Our `DefaultColumnAnalyzer` currently looks like this:

```
  /**
   * This is an Analyzer for processing default column values using built-in functions only.
   */
  object DefaultColumnAnalyzer extends Analyzer(
    new CatalogManager(BuiltInFunctionCatalog, BuiltInFunctionCatalog.v1Catalog)) {
  }
```

We could use a different catalog manager that always throws errors there specifically when we exercise this `getExistenceDefaultValues` method. This would make sure we're not inadvertently doing catalog manager lookups there anymore.

Reply:
@dtarima yea i was back and forth on this, I had this test originally but removed it in https://github.com/apache/spark/pull/49840/commits/0491833ce1f43d3555f7e7969b0dceb1fe3718fc   (hope the link works).  It seemed a little intrusive on production code and debatable value, but Im open.